From 5d6868d581290ca35df04f3d4858116c8ba20ab2 Mon Sep 17 00:00:00 2001
From: Xu Liang <liang.xu@intel.com>
Date: Wed, 21 Oct 2020 11:39:36 +0800
Subject: [PATCH] PONRTSYS-8089: GSW API - set default overhead for shaper and
 meter. The overhead of shaper may be overwritten by other application
 (PONRTSYS-7499).

---
 drivers/net/ethernet/lantiq/switch-api/gsw_flow_core.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/net/ethernet/lantiq/switch-api/gsw_flow_core.c b/drivers/net/ethernet/lantiq/switch-api/gsw_flow_core.c
index ea35db78b9f1..ee67bd076f50 100644
--- a/drivers/net/ethernet/lantiq/switch-api/gsw_flow_core.c
+++ b/drivers/net/ethernet/lantiq/switch-api/gsw_flow_core.c
@@ -3581,6 +3581,12 @@ void *ethsw_api_core_init(ethsw_core_init_t *ethcinit)
 		}
 	}
 
+	/* Set default overhead (preamble) for rate shaper and meter */
+	gsw_w32(cdev, BM_OVERHD_OVERHD_OFFSET, BM_OVERHD_OVERHD_SHIFT,
+		BM_OVERHD_OVERHD_SIZE, 0xEC);
+	gsw_w32(cdev, PCE_OVERHD_OVERHD_OFFSET, PCE_OVERHD_OVERHD_SHIFT,
+		PCE_OVERHD_OVERHD_SIZE, 20);
+
 	/** Parser Micro Code Init */
 #if defined(CONFIG_USE_EMULATOR)
 	gsw_w32(cdev, PCE_GCTRL_0_MC_VALID_OFFSET,
